designs/2021-init-command-eslint-cli/README.md

```diff
@@ -0,0 +1,136 @@
+- Repo: [eslint/eslint](https://github.com/eslint/eslint)
+- Start Date: 2020-06-29
+- RFC PR: https://github.com/eslint/rfcs/pull/79
+- Authors: Aniketh Saha ([anikethsaha](https://github.com/anikethsaha)) aladdin-add(weiran.zsd@outlook.com)
+
+# Move --init flag into a separate utility
+
+## Summary
+
+- Move the `init` command from main repo ([eslint](https://github.com/eslint/eslint)) to a new repo named `@eslint/create-config`.
+- Remove the auto-config.
+
+## Motivation
+
+Currently the whole `init` command is being shipped with the cli in the main repo. Though its not that much of size (roughly ~0.6MB for the [eslint/lib/init](https://github.com/eslint/eslint/tree/master/lib/init)), this command is not a type of command that we need everyday or everytime running eslint. It is mainly used when creating a new project or adding eslint to a project for the first time. So if we move this to a separate package that is meant to use in the command line,
+We will make use of tools like `npx` to simply run `npm init @eslint/config` or using `npm` to run `npx  @eslint/create-config` or using `yarn` to run `yarn create @eslint/config` single time for a project instead of having it in the core project.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+The implementation is mainly migrating the code from main repo to `@eslint/create-config`.
+
+The approach would be following these steps
+
+### Move `eslint --init` related files to a separate repo
+
+- Move the `eslint/lib/init/*` to the new repo's `lib/*` directory
+- Add the following `dependencies` in `package.json`(the last 2 dependencies will be removed in next step)
+
+  - `enquirer`
+  - `progress`
+  - `semver`
+  - `lodash`
+  - `json-stable-stringify-without-jsonify`
+  - `cross-spawn`
+  - `debug`
+  - `@eslint/eslintrc`
+  - `eslint` (to be removed)
+  - `espree` (to be removed)
+
+- for all those modules coming inside `../**/*`, it would be replaced using `eslint/**/*`
```

Comment:
When we implement the `exports` from [RFC80](https://github.com/eslint/rfcs/pull/80), will that prevent all of these from working, or do we no longer need those `../` requires once we remove autoconfig?

Reply:
Yes it would. I’d be curious what is actually needed after auto config is removed. Fundamentally, all we are doing is asking questions and then writing a file. It can probably be much simpler without autoconfig